doc/man3/EVP_MAC.pod

```diff
@@ -87,179 +101,168 @@ through diverse controls.
-database by name.
+EVP_MAC_provider() returns the provider that holds the implementation
+of the given I<mac>.
 
-EVP_get_macbynid() fetches a MAC implementation from the object
-database by numeric identity.
+EVP_MAC_do_all() and EVP_MAC_do_all_sorted() both traverse all known
+MAC implementations in all known providers, and call the function
+I<fn> for each of them.
+I<arg> is also passed untouched to I<fn>.
+EVP_MAC_do_all_sorted() makes sure the MACs are traversed sorted by
+name.
 
-EVP_get_macbyobj() fetches a MAC implementation from the object
-database by ASN.1 OBJECT (i.e. an encoded OID).
+=head1 PARAMETER NAMES
 
-=head1 CONTROLS
-
-The standard controls are:
+The standard parameter names are:
 
 =over 4
 
-=item B<EVP_MAC_CTRL_SET_KEY>
+=item "key"
 
-This control expects two arguments: C<unsigned char *key>, C<size_t keylen>
+This parameter should be of type B<OSSL_PARAM_OCTET_STRING>
 
-These will set the MAC key from the given string of the given length.
-The string may be any bitstring, and can contain NUL bytes.
+Its value is the MAC key as an array of bytes.
 
 For MACs that use an underlying computation algorithm, the algorithm
-I<must> be set first, see B<EVP_MAC_CTRL_SET_ENGINE>,
-B<EVP_MAC_CTRL_SET_MD> and B<EVP_MAC_CTRL_SET_CIPHER> below.
+must be set first, see parameter names "algorithm" below.
 
-=item B<EVP_MAC_CTRL_SET_IV>
+=item "iv"
 
-This control expects two arguments: C<unsigned char *key>, C<size_t keylen>
+This parameter should be of type B<OSSL_PARAM_OCTET_STRING>
 
-Some MAC implementations require an IV, this control sets the IV.
+Some MAC implementations require an IV, this parameter sets the IV.
 
-=item B<EVP_MAC_CTRL_SET_CUSTOM>
+=item "custom"
 
-This control expects two arguments: C<unsigned char *custom>, C<size_t customlen>
+This parameter should be of type B<OSSL_PARAM_OCTET_STRING>
 
 Some MAC implementations (KMAC, BLAKE2) accept a Customization String,
-this control sets the Customization String. The default value is "".
+this parameter sets the Customization String. The default value is "".
 
-=item B<EVP_MAC_CTRL_SET_SALT>
+=item "salt"
 
-This control expects two arguments: C<unsigned char *salt>, C<size_t saltlen>
+This parameter should be of type B<OSSL_PARAM_OCTET_STRING>
 
 This option is used by BLAKE2 MAC.
 
-=item B<EVP_MAC_CTRL_SET_XOF>
+=item "xof"
+
+This parameter should be of type B<OSSL_PARAM_INTEGER>
 
-This control expects one argument: C<int xof>
+It's a simple flag, the value 0 or 1 are expected.
 
 This option is used by KMAC.
 
-=item B<EVP_MAC_CTRL_SET_FLAGS>
+=item "flags"
 
-This control expects one argument: C<unsigned long flags>
+This parameter should be of type B<OSSL_PARAM_INTEGER>
 
 These will set the MAC flags to the given numbers.
 Some MACs do not support this option.
 
-=item B<EVP_MAC_CTRL_SET_ENGINE>
+=item "engine"
 
-=item B<EVP_MAC_CTRL_SET_MD>
+=item "md"
 
-=item B<EVP_MAC_CTRL_SET_CIPHER>
+=item "digest"
+
+=item "cipher"
+
+=item "algorithm"
```

Comment:
The "properties" parameter should be described here

Reply:
I have removed property params for now, need to have another look at surrounding code.  They will come back.